### ISSUE TITLE
subscribe example doesn't compile

### ISSUE THREAD

**Issue by holla2040** [electric-sheep-co/arduino-redis#51]
redis 2.4.4 installed via library manager, Arduino IDE 1.8.19 on Ubuntu 20.04.

when compiling, I get this error
no matching function for call to 'Redis::startSubscribing(subscriberLoop(std::function<void()>)::<lambda(Redis*, String, String)>, subscriberLoop(std::function<void()>)::<lambda(Redis*, RedisMessageError)>)'

    auto subRv = redis.startSubscribing(
        [=](Redis *redisInst, String channel, String msg) {
            Serial.printf("Message on channel '%s': \"%s\"\n", channel.c_str(), msg.c_str());

**Comment by holla2040** [comment]
I moved the lambda functions out of redis.startSubscribing, made them regular functions, then it compiiled.

**Comment by rpj** [comment]
Hello @holla2040, thank you very much for your detailed report & doubly so for reporting back with your fix as well!

Would you like to submit this fix as a pull request so as to receive the credit you're due? No worries whatsoever if not, I completely understand, and worry not: if you prefer not to submit the fix yourself, I'll be sure to do it, this is a necessary fix.

Thank you again!

**Comment by holla2040** [comment]
Haven't got to this yet, can I still do a pull request?

**Comment by edfletcher** [comment]
@holla2040 Absolutely, any time! I haven't gotten to it myself yet either 😄 Thank you!

**Comment by rpj** [comment]
@holla2040 Absolutely, any time! I haven't gotten to it myself yet either 😄 Thank you!

**Comment by cactixxx** [comment]
How is the pull request going?

**Comment by holla2040** [comment]
I'll work on this almost immediately.

**Comment by rpj** [comment]
Thank you @holla2040! And to make it clear: there is no time pressure, your contributions are appreciated whenever you're able to make them.

[I need to add](https://github.com/electric-sheep-co/arduino-redis/issues/55) a proper contributor & community guideline document to the project, but if it already existed it would also note that **nonconstructive** comments such as one that is nothing but a request for a status update will be summarily removed, are not representative of this project and accordingly you should not feel obligated to respond to them. I apologize for that: I'll do my best to moderate those more quickly next time.

**Comment by holla2040** [comment]
I saw that.  I need to make a non-blocking version of startSubscribing, expect this as a different pull request.

**Comment by holla2040** [comment]
OK, I followed this https://opensource.com/article/19/7/create-pull-request-github, you should now see a PR.

**Comment by holla2040** [comment]
posted a PR a few days ago.

**Comment by rpj** [comment]
@holla2040 Thank you so much! Please accept my ssincere apologies for not getting to the review yet: unfortunately I caught Covid at the Thanksgiving holiday and have been down for the count for days. I'm starting to feel better today but unfortunately have a significant work backlog to catch up on too, so my best guess is that I'll be able to get to it this weekend (hopefully sooner but wanted to give a realistic estimate). Again, my apologies as well as my appreciation for your contributions here.

**Comment by holla2040** [comment]
No problem, glad you're feeling better.
